libs/libc/unistd/lib_sethostname.c

```diff
@@ -116,8 +116,8 @@ int sethostname(FAR const char *name, size_t size)
    */
 
   flags = enter_critical_section();
-  strncpy(g_hostname, name, MIN(HOST_NAME_MAX, size));
-  g_hostname[HOST_NAME_MAX] = '\0';
+  strncpy(g_hostname, name, MIN(HOST_NAME_MAX, namelen));
```

Comment:
should we change to strlcpy?
https://stackoverflow.com/questions/6987217/strncpy-or-strlcpy-in-my-case
Actually, I guess many place which call strncpy may suffer the similar issue. So, it will be better to replace all occurrence.

Reply:
Here is the patch for fs: https://github.com/apache/incubator-nuttx/pull/5108

Reply:
Done, and squashed the commits